### PR TITLE
Add rename2 for snakebite

### PR DESCRIPTION
**Pull request by Tarrasch**
Note that the tests definetly should fail. As `snakebite 2.5.0` is not even released yet.

I wanted to push this out for review still and I checked that the tests I added passed with my local snakebite version with `rename_dont_move()` support.

### CODE REVIEW

**Comment by Tarrasch** [comment]
Basically this need to be accepted by the snakebite guys: https://github.com/spotify/snakebite/pull/128

**Comment by Tarrasch** [comment]
-3, Add `rename_dont_move` to the Cascading client or all hell will break loose.

**Comment by Tarrasch** [comment]
So. Now you guys can review this. @erikbern @plinton @ulzha

File: test/hdfs_test.py [hunk 1]
```diff
@@ -32,6 +33,43 @@ class TestException(Exception):
 
 
 @attr('minicluster')
+class ConfigurationTest(MiniClusterTestCase):
+
+    def tezt_rename_dont_move(self, client):
```
Comment (erikbern) [inline test/hdfs_test.py]: haha tezt? I have a minor preference for something less confusing like just `def rename_dont_move`

**Comment by erikbern** [comment]
LGTM

File: test/hdfs_test.py (hunk 1 above)
Comment (Tarrasch) [inline test/hdfs_test.py]: Hmm, it's so much like a `test_`, but it's not as I want to run it once for each configuration. Hmm..

**Pull request merged**